### ISSUE TITLE
Authenticating accounts if vscode.dev is blocked

### ISSUE THREAD

**Issue by zbs** [microsoft/vscode#158003]
My company blocks vscode.dev, which blocks me from authenticating my Microsoft account because it requires opening
```
https://vscode.dev/redirect?code=<code>&state=vscode%253A%252F%252Fvscode.microsoft-authentication%253FwindowId%253D1%2526nonce%253D99tYfqfqqdiZzXx1zktcFQ%2525253D%2525253D&session_state=9da792d1-1466-4b30-b184-3f517a17b8c0
```
are there any workarounds? Are there older versions of VS code that don't use vs code dev that I could use?

**Comment by zbs** [comment]
Will give that a shot. I noticed that GitHub auth also goes through vscode.dev. Is that expected?

**Comment by TylerLeonhardt** [comment]
> I noticed that GitHub auth also goes through vscode.dev. Is that expected?

Yes. That is expected. `vscode.dev` or rather `vscode.dev/redirect` is the mechanism we use to redirect back to whatever VS Code started the flow (be it Desktop, vscode.dev, Codespaces, whatever).

GitHub auth has the extra feature of a 3rd mode of auth which doesn't use vscode.dev/redirect which is the device code flow. However, since it requires more work for the user, it's the 3rd one and not the first or second.

**Comment by pmarreck** [comment]
I run Firefox Nightly and this functionality, which used to work fine on my browser and my OS (NixOS), is now broken, which means that I cannot use the Github Copilot service I pay for with my preferred browser

**Comment by pmarreck** [comment]
Actually, it just provided another way (a few-digit code) and that worked.
